src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionDescriptorBuilder.cs

```diff
@@ -727,8 +727,16 @@ private static bool IsAttributeRoutedAction(ControllerActionDescriptor actionDes
             {
                 var routeTemplate = action.AttributeRouteInfo?.Template ?? nullTemplate;
 
-                var verbs = action.ActionConstraints.OfType<HttpMethodActionConstraint>().FirstOrDefault()?.HttpMethods;
-                var formattedVerbs = string.Join(", ", verbs.OrderBy(v => v, StringComparer.Ordinal));
+                var verbs = action.ActionConstraints?
+                    .OfType<HttpMethodActionConstraint>()
+                    .FirstOrDefault()?
+                    .HttpMethods;
```

Comment:
Was the previous statement actually incorrect? I.e. was `ActionConstraints` really `null` in the failing case (or was it just that no defined constraints had type `HttpMethodActionConstraint`?